### ISSUE TITLE
Widget focus highlighting for iOS/iPadOS

### ISSUE THREAD

**Issue by xster** [flutter/flutter#53456]
Sister bug to https://github.com/flutter/flutter/issues/43365 for iPad keyboard support since it's different from macOS. 

This also raises the question of whether we should attempt to detect platform in Cupertino and have different highlighting behavior depending on the platform. 

However, figuring out what the focus highlight spec should be is non-trivial. After testing a bit, iPadOS (13.4) doesn't really have a (consistent) focus highlight handling strategy. It's also definitely different from macOS. 

![QONX7829](https://user-images.githubusercontent.com/156888/77811118-8ee31800-7055-11ea-872b-fd51125c6e12.gif)

The task switcher looks the most polished. Springboard quick search looks similar, so does Siri search results. Based on our experience with other glass-frosty things like the action sheet, it's likely a combination of overlay and color dodge.

![EMVU4660](https://user-images.githubusercontent.com/156888/77811156-dec1df00-7055-11ea-9fd9-10bc955aadbe.gif)

UITableView has a slightly different style. It's just a simple overlay. Note it's not super useful. You can't focus on anything else than the master list. And text field is the only other thing focusable. Pressing tab just inserts tabs. 

![GECY6155](https://user-images.githubusercontent.com/156888/77811190-12046e00-7056-11ea-9e3d-2f68e9aed45f.gif)

WebKit itself has yet another style. The simplest. A blue box.

**Comment by xster** [comment]
This is worth revisiting in iOS 14 to see if the behaviors consolidate more than the current state.

**Comment by jmagman** [comment]
I think parts of this are being tracked with https://github.com/flutter/flutter/issues/55809, that one is newer though.

**Comment by justinmc** [comment]
CC @MitchellGoodwin
